src/tokens/tokens.listener.ts

```diff
@@ -62,21 +62,17 @@ export class TokenListener implements EventListener {
     private blockchain: BlockchainConnectorService,
   ) {}
 
-  async onEvent(subName: string, event: Event, process: EventProcessor) {
+  async onEvent(subName: string, event: Event) {
     const signature = this.trimEventSignature(event.signature);
     switch (signature) {
       case tokenCreateEventSignature:
-        process(await this.transformTokenPoolCreationEvent(event));
-        break;
+        return await this.transformTokenPoolCreationEvent(event);
```

Comment:
`return await` is redundant here since you just want to forward the promise along (you can just `return`)

Reply:
Addressed in ac69f15a4cde993d17f08878e3c0c9cfcbeeea77